src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,119 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 1 << 4;
+    private static final int MAX_CAPACITY = 1 << 30;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        if (table == null) {
+            table = new Node[INITIAL_CAPACITY];
+        }
+        int i = index(hash(key));
+        Node<K, V> currentNode = table[i];
+        int threshold = (int) (table.length * LOAD_FACTOR);
+        if (currentNode == null) {
+            table[i] = newNode;
+            if (++size > threshold) {
+                resize();
+            }
```

Comment:
could we make this check only ones in this method (at the beginning for example)

Reply:
Yes, we can, but in cases where we have the same key, but just the value is replaced, then the size check will be redundant, so I check for resize only the places where the size increases. In order to improve performance.

Reply:
This will not significantly affect the performance, and the code will be more readable, so I would put the test at the beginning, but you can leave it as is